packages/components/pie-cookie-banner/test/helpers/page-object/pie-cookie-banner.page.ts

```diff
@@ -0,0 +1,282 @@
+    private readonly bodyNecessaryOnlyLinkLocator;
+    private readonly bodyManagePreferencesLinkLocator: Locator;
+    private readonly bodyCookieStatementLinkLocator: Locator;
+    private readonly bodyCookieTechnologiesLinkLocator: Locator;
+    private readonly modalDescriptionLocator: Locator;
+
+    constructor (page: Page) {
+        super(page);
+        this.componentLocator = page.getByTestId(cookieBanner.selectors.container.dataTestId);
+        this.descriptionLocator = page.getByTestId(cookieBanner.selectors.description.dataTestId);
+        this.acceptAllButtonLocator = page.getByTestId(cookieBanner.selectors.acceptAllButton.dataTestId);
+        this.necessaryOnlyButtonLocator = page.getByTestId(cookieBanner.selectors.necessaryOnlyButton.dataTestId);
+        this.managePrefsLinkLocator = page.getByTestId(cookieBanner.selectors.managePreferencesLink.dataTestId);
+        this.bodyAcceptAllLinkLocator = page.getByTestId(cookieBanner.selectors.bodyAcceptAllLink.dataTestId);
+        this.bodyNecessaryOnlyLinkLocator = page.getByTestId(cookieBanner.selectors.bodyNecessaryOnlyLink.dataTestId);
+        this.bodyManagePreferencesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyManagePreferencesLink.dataTestId);
+        this.bodyCookieStatementLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieStatementLink.dataTestId);
+        this.bodyCookieTechnologiesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieTechnologiesLink.dataTestId);
+        this.modalDescriptionLocator = page.getByTestId(cookieBanner.selectors.modalDescription.dataTestId);
+    }
+
+    /**
+     * Checks whether the cookie banner is currently visible on the page.
+     *
+     * @returns {Promise<boolean>} A Promise that resolves to a boolean value indicating
+     *                             whether the cookie banner is visible (`true`) or not (`false`).
+     */
+    async isCookieBannerVisible () : Promise<boolean> {
+        return this.componentLocator.isVisible();
+    }
+
+    /**
+     * Clicks the "Accept All" button.
+     *
+     * @returns {Promise<void>} A Promise that resolves once the "Accept All" button
+     *                          has been successfully clicked.
+     */
+    async clickAcceptAllButton () : Promise<void> {
+        await this.acceptAllButtonLocator.click();
+    }
+
+    /**
+     * Clicks the "Accept All" link within the body.
+     *
+     * @returns {Promise<void>} A Promise that resolves once the "Accept All" link
+     *                          within the body has been successfully clicked.
+     */
+    async clickBodyAcceptAllLink () : Promise<void> {
+        await this.bodyAcceptAllLinkLocator.click();
+    }
+
+    /**
+     * Clicks either the "Accept All" button or the "Accept All" link within the body, based on the specified element level.
+     *
+     * @param {string} elementLevel The level of the element to click, either 'action' or 'body'.
```

Comment:
This is an excellent opportunity to use a string union type! It could be `action | body`

[reference](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html)